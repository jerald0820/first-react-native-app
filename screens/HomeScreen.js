import { StatusBar } from 'expo-status-bar';
import { StyleSheet, TouchableOpacity, View, Text } from 'react-native';
import globalStyle from '../style/global';

export default function HomeScreen({ navigation }) {
  return (
    <View style={styles.container}>
      <Text style={globalStyle.headLine}>Welcome to Your Test App</Text>
      <View style={globalStyle.buttonGroup}>
        <TouchableOpacity 
          style={globalStyle.primaryButton}
          onPress={() => navigation.navigate("ProfileScreen", { language: "french", name: "John Doe" })}
        >
          <Text style={globalStyle.primaryButtonText}>Profile french</Text>
        </TouchableOpacity>
        <TouchableOpacity 
          style={globalStyle.primaryButton}
          onPress={() => navigation.navigate("ProfileScreen", { language: "english", name: "John Doe" })}
        >
          <Text style={globalStyle.primaryButtonText}>Profile english</Text>
        </TouchableOpacity>
      </View>
      <StatusBar style="auto" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  }
});
