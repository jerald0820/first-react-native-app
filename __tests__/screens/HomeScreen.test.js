import renderer from "react-test-renderer";

import HomeScreen from '../../screens/HomeScreen';

describe("<HomeScreen />", () => {
    it("has 2 child", async () => {
        const tree = renderer.create(<HomeScreen />).toJSON();
        expect(tree.children.length).toBe(2); // 2 Button Components
    });

    it("renders correctly", async () => {
        const tree = renderer.create(<HomeScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});