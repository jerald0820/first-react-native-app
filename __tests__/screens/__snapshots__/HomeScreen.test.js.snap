// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 24,
        "fontWeight": 700,
        "marginBottom": 50,
      }
    }
  >
    Welcome to Your Test App
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "columnGap": 5,
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#2B4D66",
          "borderRadius": 15,
          "opacity": 1,
          "padding": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "#BED8FB",
          }
        }
      >
        Profile french
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#2B4D66",
          "borderRadius": 15,
          "opacity": 1,
          "padding": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "#BED8FB",
          }
        }
      >
        Profile english
      </Text>
    </View>
  </View>
</View>
`;
